okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -13,1331 +13,1173 @@
+      throw RuntimeException(e) // Unexpected!
     }
   }
 
   /**
-   * Returns this URL as a {@link URI java.net.URI}. Because {@code URI} is more strict than this
-   * class, the returned URI may be semantically different from this URL:
+   * Returns this URL as a [java.net.URI][URI]. Because `URI` is more strict than this class, the
+   * returned URI may be semantically different from this URL:
+   *
+   *  * Characters forbidden by URI like `[` and `|` will be escaped.
    *
-   * <ul>
-   *     <li>Characters forbidden by URI like {@code [} and {@code |} will be escaped.
-   *     <li>Invalid percent-encoded sequences like {@code %xx} will be encoded like {@code %25xx}.
-   *     <li>Whitespace and control characters in the fragment will be stripped.
-   * </ul>
+   *  * Invalid percent-encoded sequences like `%xx` will be encoded like `%25xx`.
    *
-   * <p>These differences may have a significant consequence when the URI is interpreted by a
-   * webserver. For this reason the {@linkplain URI URI class} and this method should be avoided.
+   *  * Whitespace and control characters in the fragment will be stripped.
+   *
+   * These differences may have a significant consequence when the URI is interpreted by a
+   * web server. For this reason the [URI class][URI] and this method should be avoided.
    */
-  public URI uri() {
-    String uri = newBuilder().reencodeForUri().toString();
+  fun uri(): URI {
+    val uri = newBuilder().reencodeForUri().toString()
     try {
-      return new URI(uri);
-    } catch (URISyntaxException e) {
+      return URI(uri)
+    } catch (e: URISyntaxException) {
       // Unlikely edge case: the URI has a forbidden character in the fragment. Strip it & retry.
       try {
-        String stripped = uri.replaceAll("[\\u0000-\\u001F\\u007F-\\u009F\\p{javaWhitespace}]", "");
-        return URI.create(stripped);
-      } catch (Exception e1) {
-        throw new RuntimeException(e); // Unexpected!
+        val stripped = uri.replace(Regex("[\\u0000-\\u001F\\u007F-\\u009F\\p{javaWhitespace}]"), "")
+        return URI.create(stripped)
+      } catch (e1: Exception) {
+        throw RuntimeException(e) // Unexpected!
       }
     }
   }
 
-  /** Returns either "http" or "https". */
-  public String scheme() {
-    return scheme;
-  }
-
-  public boolean isHttps() {
-    return scheme.equals("https");
-  }
+  /** Returns either "http" or "https".  */
+  fun scheme(): String = scheme
 
   /**
    * Returns the username, or an empty string if none is set.
    *
-   * <p><table summary="">
-   *   <tr><th>URL</th><th>{@code encodedUsername()}</th></tr>
-   *   <tr><td>{@code http://host/}</td><td>{@code ""}</td></tr>
-   *   <tr><td>{@code http://username@host/}</td><td>{@code "username"}</td></tr>
-   *   <tr><td>{@code http://username:password@host/}</td><td>{@code "username"}</td></tr>
-   *   <tr><td>{@code http://a%20b:c%20d@host/}</td><td>{@code "a%20b"}</td></tr>
+   * <table summary="">
```

Comment:
Would be nice to convert those into Markdown tables, follow-up

Reply:
Good idea.